torch/_dynamo/variables/torch.py

```diff
@@ -367,58 +367,44 @@ def call_function(
             return ConstantVariable(
                 torch.backends.cudnn.is_acceptable(tensor_inp), **options
             )
-        if (
-            self.value.__name__ == "get_state"
-            and hasattr(self.value, "__self__")
-            and isinstance(self.value.__self__, torch._C.Generator)
-        ):
-
-            def get_state_from_generator():
-                return self.value()
+        elif self.value in (torch.random.get_rng_state, torch.cuda.get_rng_state):
+            # Insert the get rng state ops directly into the graphs. AOT
+            # Autograd during functionalization will use these ops to track the
+            # rng state and functionalize RNG ops.
+            example_value = self.value()
+
+            # Generate source - The tensor here is generated inside the program,
+            # so we don't have to get the source. But Dynamo logic assumes that
+            # each Tensor has a source. So, lets get a special source -
+            # GenerateStrateSource.
+            device = "cpu" if self.value is torch.random.get_rng_state else "cuda"
+            seed = (
+                torch.initial_seed() if device == "cpu" else torch.cuda.initial_seed()
+            )
+            source = GeneratorStateSource(device, seed)
 
             return wrap_fx_proxy(
                 tx=tx,
                 proxy=tx.output.create_proxy(
                     "call_function",
-                    get_state_from_generator,
+                    self.value,
                     *proxy_args_kwargs(args, kwargs),
                 ),
-                example_value=self.value(),
-                source=GeneratorStateSource(
-                    self.value.__self__.device.type, self.value.__self__.initial_seed()
-                ),
+                example_value=example_value,
+                source=source,
                 **options,
             )
-        if (
-            self.value.__name__ == "set_state"
-            and hasattr(self.value, "__self__")
-            and isinstance(self.value.__self__, torch._C.Generator)
-        ) or self.value == torch.random.set_rng_state:
-            assert len(args) == 1
-            assert isinstance(args[0], TensorVariable)
-
-            unimplemented(
-                "TODO: make torch.random.set_rng_state work with FakeTensor/aot_autograd"
-            )
```

Comment:
Oh, this would not work with AOT Autograd backend yet (not sure about segfaults, but it wont really respect the get/set_rng_state, because they aren't visible to AOT Autograd). To fix that, we will need functionalization of RNG ops (which I am working on).

Currently, we don't have any unit/model tests with get/set_rng_state that uses Dynamo + AOT Autograd. And thats why tests are passing.

This PR allows me to run torch.compile on toy models for functionalization of RNG ops work. I am ok to raise an execption though if thats better.

Reply:
My point is, if the backend is not ready, then removing the unimplemented (which induces a graph break) is worse because it makes something that previously worked (via graph break) into something that crashes at compile time.

Reply:
Yes. Let me graph break for now w/ the appropriate warning.